src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -4864,23 +4853,6 @@ JCDiagnostic getDiagnostic(DiagnosticType dkind, DiagnosticPosition pos, Symbol
         }
     }
 
-    /**
-     * BadConstructorReferenceError error class indicating that a constructor reference symbol has been found,
-     * but pointing to a class for which an enclosing instance is not available.
-     */
-    class BadConstructorReferenceError extends InvalidSymbolError {
```

Comment:
This was no longer needed. We had a duplication between a check issued when resolving method references (see `ConstructorReferenceLookupHelper`) and a post-resolution check in `Attr::visitReference`. I think keeping the latter makes much more sense, as that brings more consistency between the code paths for instance creation expression and constructor reference expression.